src/libraries/Microsoft.PowerFx.Core/Entities/External/InternalTableParameters.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Entities.Delegation;
+using Microsoft.PowerFx.Core.Entities.QueryOptions;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Functions.Delegation.DelegationMetadata;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    internal class InternalTableParameters : IExternalTabularDataSource
+    {
+        public const string IsChoiceValue = "Value";
+
+        public readonly Dictionary<string, string> ColumnsWithRelationships;
+
+        public InternalTableParameters(AggregateType recordType, DisplayNameProvider displayNameProvider, TableDelegationInfo tableParameters)
+        {
+            string GetDisplayName(string fieldName) => displayNameProvider == null || !displayNameProvider.TryGetDisplayName(new DName(fieldName), out DName displayName) ? fieldName : displayName.Value;
+
+            EntityName = new DName(tableParameters.TableName);
+            IsWritable = !tableParameters.IsReadOnly;
+            _tableParameters = tableParameters;
+            _type = recordType._type;
+            RecordType = (RecordType)recordType;
+
+            IEnumerable<string> fieldNames = displayNameProvider.LogicalToDisplayPairs.Select(pair => pair.Key.Value);
+
+            _displayNameMapping = new BidirectionalDictionary<string, string>(fieldNames.Select(f => new KeyValuePair<string, string>(f, GetDisplayName(f))).ToDictionary(kvp => kvp.Key, kvp => kvp.Value));
+            _externalDataEntityMetadataProvider = new InternalDataEntityMetadataProvider();
+            _externalDataEntityMetadataProvider.AddSource(Name, new InternalDataEntityMetadata(tableParameters.TableName, tableParameters.DatasetName, _displayNameMapping));
+            _externalTableMetadata = new InternalTableMetadata(RecordType, Name, Name, tableParameters.IsReadOnly);
+            _delegationMetadata = new DelegationMetadataBase(_type, new CompositeCapabilityMetadata(_type, GetCapabilityMetadata(recordType, tableParameters)));
+            _tabularDataQueryOptions = new TabularDataQueryOptions(this);
+            _previousDisplayNameMapping = null;
+
+            ColumnsWithRelationships = tableParameters.ColumnsWithRelationships;
+        }
+
+        public TableDelegationInfo TableParameters => _tableParameters;
```

Comment:
>TableDelegationInfo [](http://example.com/codeflow?start=15&length=19)

This seems to be completely different than DelegationMetadataBase

Reply:
The goal of this PR here is to make sure we can execute delegation code for
- CDP connectors (in Fx.Connectors assembly but can be tested in DV repo where delegation code is)
DV PR [https://github.com/microsoft/Power-Fx-Dataverse/pull/545](https://github.com/microsoft/Power-Fx-Dataverse/pull/545)
- DV - we should make sure we don't break. Most of DV PR is about using IDelegationMetadata (line 57 here)
- non-CDP but tabular connectors like ServiceNow - NL PR [https://msazure.visualstudio.com/OneAgile/_git/Power%20Fx%20Internal/pullrequest/10862241](https://msazure.visualstudio.com/OneAgile/_git/Power%20Fx%20Internal/pullrequest/10862241)
See ServiceNowDelegationInfo class in ServiceNowRecordType (we only use defaults here but the goal is to allow customers to write their own tabular connector and provide an equivalent of x-ms-capabilities, x-ms-relationships/relatedEntities in TableDelegationInfo 

What's probably creating confusion is "table delegation" and "table parameters" due to a last minute rename Mike wanted
I'll realign names ...